pages/owned/index.vue

```diff
@@ -55,18 +58,32 @@ onMounted(async () => {
   if (!wallet.value) {
     return;
   }
-  fetchOwnedNFTs();
+  fetchOwnedAndListingNFTs();
 })
 
 watch(wallet, (newWallet) => {
-  if (newWallet) fetchOwnedNFTs();
+  if (newWallet) fetchOwnedAndListingNFTs();
 })
 
 
 const { connect } = store;
 
-async function fetchOwnedNFTs() {
-  ownedList.value = await queryOwnedNFTClasses(wallet.value);
+function makeKey(classId: string, nftId: string) {
+  return `${classId}_${nftId}`;
+}
+
+async function fetchOwnedAndListingNFTs() {
+  const ownedList = await queryOwnedNFTClasses(wallet.value);
+  ownedList.forEach((n) => {
+    ownedMap.value[makeKey(n.classId, n.id)] = n;
+  });
+
+  const listingEvents = await getRecentListingEvents(wallet.value);
```

Comment:
Can we use `https://github.com/likecoin/likecoin-chain-tx-indexer/pull/64#issue-1508273452` instead? Using the current `create_listing` query we actually don't know if the listing is already bought or still valid